### DISCUSSION TITLE
Is it possible to call extension functions?

### DISCUSSION THREAD

**Discussion by wozeparrot** [pyopencl/pyopencl#696] [Q&A]
Basically support for `clGetExtensionFunctionAddressForPlatform`, our usecase in [tinygrad](https://github.com/tinygrad/tinygrad) is for the `cl_amd_copy_buffer_p2p` which has an extension function `clEnqueueCopyBufferP2PAMD`

Currently it is possible by maintaining a fork of pyopencl, but it's better if we can just use upstream. I'm not sure what the policy is on upstreaming a change like this because it is very extension/platform specific.

**Comment by inducer** [comment]
Sure, I'd be happy to look at a PR along those lines. Wrapping CL extensions is definitely in scope. There are a few precedents for that in the code base even today. The one caveat is that the corresponding header has to be available at build time (or part of the shipped extension header), and the wrapper must be sufficiently `#ifdef`'d to ensure that the build succeeds irrespective of whether the extension header is available.
